### PR TITLE
Adding MasterMind service manager docs

### PR DESCRIPTION
**Pull request by brunograz**

### CODE REVIEW

File: docs/admin/service_manager.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+## Currently supported API endpoints - MasterMind Service Manager
+
+* `POST /v1/stack` - Create a new stack deploying services, creating networks,
+  volumes, secrets and configs defined in the docker-compose file.
+
+* `POST /v1/stack/{name}` - Return the status of an existing stack with the
+  services currently running in the Stack and the total amount of replicas.
+
+* `POST /v1/stack/delete/{name}` - Delete the stack removing services and
```
Comment (taliaga) [inline docs/admin/service_manager.md:43]: I'd have left the same `v1/stack/{name}` but changing the http verb to delete. Not a change request, just sharing what I've seen most often :)

File: docs/admin/service_manager.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  networks. Note that volumes, secrets and configs are not deleted.
+
+![image alt text](../images/swagger_example.png)
+
+The above image depicts a `POST` request to the `/v1/stack` endpoint. A set of
+parameters are given as strings plus a key-value mapping JSON object for this
+endpoint:
+
+* **name:** Name of the stack - this is equivalent to the stack_name parameter
+  used in a docker stack deploy command;
```
Comment (taliaga) [inline docs/admin/service_manager.md:53]: link to https://docs.docker.com/engine/swarm/stack-deploy/#deploy-the-stack-to-the-swarm would be useful

File: docs/admin/service_manager.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+* **compose-file:** A string containing the entire docker-compose file in yaml
+  format; Variables may be referenced in this file which are obtained from
+  compose-vars; The syntax used for variable substitution is analogous to
+  docker-compose, i.e. `${VARIABLE}`.
```
Comment (taliaga) [inline docs/admin/service_manager.md:58]: worth a comment on how to deal with default values ${VARIABLE:-default}

File: docs/admin/service_manager.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+* `POST /v1/stack/{name}` - Return the status of an existing stack with the
+  services currently running in the Stack and the total amount of replicas.
+
+* `POST /v1/stack/delete/{name}` - Delete the stack removing services and
+  networks. Note that volumes, secrets and configs are not deleted.
+
+![image alt text](../images/swagger_example.png)
+
+The above image depicts a `POST` request to the `/v1/stack` endpoint. A set of
```
Comment (taliaga) [inline docs/admin/service_manager.md:48]: If the service is up and running, how about deploying the swagger "live" doc and placing a link in this document?

File: docs/admin/service_manager.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+* **name:** Name of the stack - this is equivalent to the stack_name parameter
+  used in a docker stack deploy command;
+
+* **compose-file:** A string containing the entire docker-compose file in yaml
+  format; Variables may be referenced in this file which are obtained from
+  compose-vars; The syntax used for variable substitution is analogous to
+  docker-compose, i.e. `${VARIABLE}`.
+
+* **compose-vars:** Also a string in yaml format containing the a key-value
```
Comment (taliaga) [inline docs/admin/service_manager.md:60]: format?
```
key value
key: value,
```
etc.

File: docs/admin/service_manager.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+it may require specifically setting up security groups and/or other firewall
+rules to ensure these ports are accessible*.
+
+## Security
+
+In order to safely connect to a remote Docker engine, the Mastermind Service
+Manager requires TLS certificates to protect the connection to the remote
+Docker engine. [This is a common practice in Docker](https://docs.docker.com/engine/security/https/).
+
+[Rancher](http://rancher.com/) makes use of docker-machine which automatically
```
Comment (taliaga) [inline docs/admin/service_manager.md:94]: From a reader's point of view, rancher seems to come a bit out of the blue in mastermind docs. I don't recall if there's some part giving some background context or relating both. Is there?

File: docs/admin/service_manager.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+  Name of the file that used to create the secret or config.
+  The service manager will check the ‘external_files’ parameter and get
+  the file content.
+
+## Known issues
+
+There is a known bug in newer versions of Docker engine preventing the
+deployment of stacks to a Swarm cluster. The stack will be deployed but no
+actual containers will start, hanging on ‘Pending’ status. For more information
+about the bug please visit [#35093](https://github.com/moby/moby/issues/35093)
```
Comment (taliaga) [inline docs/admin/service_manager.md:188]: Seems fixed now :)

File: docs/admin/service_manager.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+  Compose and therefore won’t be created by the service manager.
+  Raises an exception in case the volume or network does not exist.
+* labels:
+  Labels to be associated to the volume or network.
+  It can be either an array or a key-value mapping JSON object.
+
+### Secrets and Configs
+
+* file:
+  Name of the file that used to create the secret or config.
```
Comment (taliaga) [inline docs/admin/service_manager.md:179]: typo: remove 'that' before used.

File: docs/admin/service_manager.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+## Docker Compose Reference - Parameters Supported by the Service Manager
+
+The Docker compose file format is quite complex, reflecting the complexity of
+container systems. The Mastermind Service Manager does not currently support
+all parameters defined in docker-compose. Below, the set of supported
+parameters is provided.
+
+### Services
+
+* command:
```
Comment (taliaga) [inline docs/admin/service_manager.md:114]: if you add these parameters in bold or as code it'll make the list more readable.

**Review by Cerfoglg — APPROVED** [review]

File: docs/admin/service_manager.md (hunk 1 above)
Comment (brunograz) [reply docs/admin/service_manager.md:43]: You mean change it to a DELETE `v1/stack/{name}`? We have tried this but the content body gets dropped when using this HTTP method. Adding those parameters to the URL would make it really large.

File: docs/admin/service_manager.md (hunk 4 above)
Comment (brunograz) [reply docs/admin/service_manager.md:48]: IMO this service should only be used internally by MasterMind API and should not have any user interaction. We could make it live in swaggerhub but with all the "Try me" options disabled.

File: docs/admin/service_manager.md (hunk 9 above)
Comment (brunograz) [reply docs/admin/service_manager.md:114]: I have tried adding it in bold but somehow the css is not applying it to lists. Anything we could help to solve this?

**Pull request merged**